### DISCUSSION TITLE
[sBTC] Open Design Questions

### DISCUSSION THREAD

**Discussion by jcnelson** [stacks-network/stacks-core#3547] [General]
* Stacker-controlled Bitcoin wallet script design
* Best way to add support for existing Bitcoin wallets for peg-ins 
* Paying for Stacks transactions with sBTC
* Stacking with sBTC, or other forms of BTC yield on sBTC?  Needs an economist's attention
* Who pays the Bitcoin tx fee for peg-out fulfillment?  How do we set this fee?
* Multiple hard-forks to speed up deployment?
* Stacker peg hand-off concerns, especially w.r.t. TOCTTOU issues with client wallets sending to a freshly-stale peg wallet address
* Delegating Stacker signing authority to a custodian?
* How might we make the signer process more accessible?

**Comment by JoseSK999** [comment]
I think the Stacker Bitcoin wallet needs to be P2TR to allow up to [11,759 signing keys](https://bitcoin.stackexchange.com/questions/114048/what-is-the-largest-multisig-quorum-currently-possible). Something similar to [this degrading multisig](https://github.com/bitcoinops/taproot-workshop/blob/master/3.1-degrading-multisig-case-study.ipynb) setup could be used, but I don't know if the MuSig key aggregation is suitable (it's interactive and a spend from it requires all the keys to participate).

**Reply by jcnelson** [reply]
We're using a variation of FROST [1] to have the Stackers collectively produce a single signature.  It doesn't matter how many signers there are; the resulting transaction takes the same amount of block space.

[1] https://github.com/Trust-Machines/frost
